lucene/core/src/java/org/apache/lucene/search/TopScoreDocCollector.java

```diff
@@ -74,11 +75,25 @@ public void collect(int doc) throws IOException {
           totalHits++;
           hitsThresholdChecker.incrementHitCount();
 
-          if (score <= pqTop.score) {
```

Comment:
I find this simpler:
`if (score <= pqTop.score && (bottomValueChecker == null || score < bottomValueChecker.getBottomValue()) {`
This is redundant for the segment that holds the current max minimum score but we can optimize this case later, when the logic is fully implemented.
When comparing the global minimum score we need to use lesser than `<` rather than `<=`. This is important since we tiebreak on document id and the global minimum score can come from a document id that is **after** this  local segment. We can also optimize this part and record the global document id that is associated with the current max score in the BottomValueChecker but that's not necessary in the first iteration IMO.

Reply:
Agreed, +1.